service/discovery/azure/compute.go

```diff
@@ -123,41 +123,71 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
 	}
 	list = append(list, virtualMachines...)
 
-	// Discover functions
-	function, err := d.discoverFunctions()
+	// Discover functions and web apps
+	function, err := d.discoverSites()
 	if err != nil {
 		return nil, fmt.Errorf("could not discover functions: %w", err)
 	}
-	list = append(list, function...)
+	if function != nil {
+		list = append(list, function...)
+	}
 
 	return
 }
 
 // Discover function
-func (d *azureComputeDiscovery) discoverFunctions() ([]voc.IsCloudResource, error) {
+func (d *azureComputeDiscovery) discoverSites() ([]voc.IsCloudResource, error) {
 	var list []voc.IsCloudResource
 
 	// initialize functions client
-	if err := d.initFunctionsClient(); err != nil {
+	if err := d.initWebAppsClient(); err != nil {
 		return nil, err
 	}
 
 	// List functions
 	err := listPager(d.azureDiscovery,
-		d.clients.functionsClient.NewListPager,
-		d.clients.functionsClient.NewListByResourceGroupPager,
+		d.clients.webAppsClient.NewListPager,
+		d.clients.webAppsClient.NewListByResourceGroupPager,
 		func(res armappservice.WebAppsClientListResponse) []*armappservice.Site {
 			return res.Value
 		},
 		func(res armappservice.WebAppsClientListByResourceGroupResponse) []*armappservice.Site {
 			return res.Value
 		},
-		func(function *armappservice.Site) error {
-			r := d.handleFunction(function)
+		func(site *armappservice.Site) error {
+			var r voc.IsCompute
+
+			// Check kind of site (see https://github.com/Azure/app-service-linux-docs/blob/master/Things_You_Should_Know/kind_property.md)
+			switch *site.Kind {
+			case "app": // Windows Web App
+			// TODO(all): TBD
+			case "app,linux": // Linux Web app
+			// TODO(all): TBD
+			case "app,linux,container": // Linux Container Web app
+			// TODO(all): TBD
```

Comment:
Add test cases for these, maybe to check that `r` is nil/empty

Reply:
That is a log of overhead because I have to add a function for each case in the Do method. I think that is out of proportion and ok if we only check 2 cases.